core/src/transport/global_only.rs

```diff
@@ -290,17 +291,17 @@ mod ipv6_global {
     }
 }
 
-impl<T> GlobalIpOnly<T> {
+impl<T> Transport<T> {
     pub fn new(transport: T) -> Self {
-        GlobalIpOnly { inner: transport }
+        Transport { inner: transport }
     }
 }
 
-impl<T: Transport + Unpin> Transport for GlobalIpOnly<T> {
-    type Output = <T as Transport>::Output;
-    type Error = <T as Transport>::Error;
-    type ListenerUpgrade = <T as Transport>::ListenerUpgrade;
-    type Dial = <T as Transport>::Dial;
+impl<T: TransportCore + Unpin> TransportCore for Transport<T> {
+    type Output = <T as TransportCore>::Output;
+    type Error = <T as TransportCore>::Error;
+    type ListenerUpgrade = <T as TransportCore>::ListenerUpgrade;
+    type Dial = <T as TransportCore>::Dial;
```

Comment:
Please prefer fully-qualifying over renaming imports.

```suggestion
impl<T: crate::Transport + Unpin> crate::Transport for Transport<T> {
    type Output = <T as crate::Transport>::Output;
    type Error = <T as crate::Transport>::Error;
    type ListenerUpgrade = <T as crate::Transport>::ListenerUpgrade;
    type Dial = <T as crate::Transport>::Dial;
```

Reply:
@thomaseizinger sure no problem, done.